lib/haml_lint/directive.rb

```diff
@@ -0,0 +1,123 @@
+module HamlLint
+  # Handles linter configuration transformation via Haml comments.
+  class Directive
+    LINTER_NAME_PATTERN = '(?:[A-Z]\w+)'.freeze
+    MULTIPLE_LINTERS_PATTERN = "(?:#{LINTER_NAME_PATTERN} , )*#{LINTER_NAME_PATTERN}".freeze
+    LINTERS_PATTERN = "(?<linters>all|#{MULTIPLE_LINTERS_PATTERN})".freeze
+
+    DIRECTIVE_REGEXP = Regexp.new(
+      (' haml-lint : (?<mode>(?:dis|en)able)\b ' + LINTERS_PATTERN).gsub(' ', '\s*')
```

Comment:
Minor nit: I had to read this a few times to understand what the `gsub` was achieving. I get it now (allowing any whitespace, including none). I wonder if it would have been more readable to just inline it though.